sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3271,205 +3298,1214 @@ atomic_update(T *p, simd<Toffset, N> byte_offset, simd_mask<N> mask,
+///   properties list. Only L1/L2 properties are used.
+///   Other properties are ignored.
+/// @return A vector of the old values at the memory locations before the
+///   update.
+template <atomic_op Op, typename T, int N, typename Toffset,
+          typename RegionTy = region1d_t<Toffset, N, 1>,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    __ESIMD_DNS::get_num_args<Op>() == 0 &&
+        ext::oneapi::experimental::is_property_list_v<PropertyListT>,
+    simd<T, N>>
+atomic_update(T *p, simd_view<Toffset, RegionTy> offsets, simd_mask<N> mask,
+              PropertyListT props = {}) {
+  return atomic_update<Op, T, N>(p, offsets.read(), mask, props);
+}
+
+/// simd<T, N>
+/// atomic_update(T *p, simd_view<Toffset, RegionTy> byte_offset,
+///               simd_mask<N> mask, props = {});               /// (usm-au0-4)
+///
+/// A variation of \c atomic_update API with \c offsets represented as
+/// \c simd_view object without mask operand.
+///
+/// @tparam Op The atomic operation - can be \c atomic_op::inc or
+///   \c atomic_op::dec, \c atomic_op::load.
+/// @tparam T The vector element type.
+/// @tparam N The number of memory locations to update
+/// @param p The USM pointer.
+/// @param byte_offset The simd_view of 32-bit or 64-bit offsets in bytes.
+/// @param props The parameter 'props' specifies the optional compile-time
+///   properties list. Only L1/L2 properties are used.
+///   Other properties are ignored.
+/// @return A vector of the old values at the memory locations before the
+///   update.
+template <atomic_op Op, typename T, int N, typename Toffset,
+          typename RegionTy = region1d_t<Toffset, N, 1>,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    __ESIMD_DNS::get_num_args<Op>() == 0 &&
+        ext::oneapi::experimental::is_property_list_v<PropertyListT>,
+    simd<T, N>>
+atomic_update(T *p, simd_view<Toffset, RegionTy> byte_offset,
+              PropertyListT props = {}) {
+  simd_mask<N> mask = 1;
+  return atomic_update<Op, T, N>(p, byte_offset.read(), mask, props);
+}
+
+/// A variation of \c atomic_update API with \c offset represented as
+/// scalar.
+///
+/// @tparam Op The atomic operation - can be \c atomic_op::inc or
 /// \c atomic_op::dec, \c atomic_op::load.
 /// @tparam T The vector element type.
 /// @tparam N The number of memory locations to update.
 /// @param p The USM pointer.
-/// @param byte_offset The scalar 32-bit or 64-bit offset in bytes.
+/// @param byte_offset The scalar 32-bit or 64-bit offset in bytes.
+/// @param mask Operation mask, only locations with non-zero in the
+///   corresponding mask element are updated.
+/// @return A vector of the old values at the memory locations before the
+///   update.
+///
+template <atomic_op Op, typename T, int N, typename Toffset>
```

Comment:
Please move this function (and other 0-source funcs) up - all functions for 0-sources op should go before the 1st atomic_update with 1-source. That will make the development/review/search easier.

Reply:
Fixed in https://github.com/intel/llvm/pull/11862/commits/687650629fde42caf2821cdcd5d7f952f6fafd8b.